packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -664,6 +661,139 @@ export class RunningSummarizer implements IDisposable {
 		});
 	}
 
+	/**
+	 * Tries to summarize with dynamic retries, i.e., retry is based on the failure params and error received.
+	 * For instance, if summarize failed due to a server error that has `retryAfterSeconds` set, wait for that duration
+	 * and retry summarize. Similarly, if summary generation failed but the error is retriable, retry before bailing out.
+	 */
+	private async trySummarizeWithDynamicRetries(
+		reason: SummarizeReason,
+		cancellationToken = this.cancellationToken,
+	) {
+		// Currently, we try 2 overall attempts with retries in each attempt. The second attempt will close the
+		// summarizer at the summarization call site but we need to keep this code for the time being. This will be
+		// reduced to a single attempt.
+		const attemptOptions: ISummarizeOptions[] = [
+			{ refreshLatestAck: false, fullTree: false },
+			{ refreshLatestAck: true, fullTree: false },
```

Comment:
Is `refreshLatestAck` supposed to be `true` or `false` here?

Reply:
It has to be true so that the code in ContainerRuntime can fetch the snapshot and then close the container. This can be removed once that code either goes away or is moved here. And then `refreshLatestAck` will not be required.

Reply:
I see now.